src/IO/H5/VolumeData.cpp

```diff
@@ -18,10 +20,61 @@
 #include "IO/H5/Header.hpp"
 #include "IO/H5/Helpers.hpp"
 #include "IO/H5/Version.hpp"
-
+#include "Utilities/Numeric.hpp"
 /// \cond HIDDEN_SYMBOLS
 namespace h5 {
-VolumeData::VolumeData(const bool exists, detail::OpenGroup&& group,
+namespace {
+// Append the element extents and connectevity to the total extents and
+// connectivity
+void append_element_extents_and_connectivity(
+    gsl::not_null<std::vector<size_t>*> total_extents,
+    gsl::not_null<std::vector<int>*> total_connectivity,
+    gsl::not_null<int*> total_points_so_far, size_t dim,
+    ExtentsAndTensorVolumeData element) {
+  // Process the element extents
+  const auto& extents = element.extents;
+  if (extents.size() != dim) {
+    ERROR("Trying to write data of dimensionality"
+          << extents.size() << "but the VolumeData file has dimensionality"
+          << dim << ".");
+  }
+  total_extents->insert(total_extents->end(), extents.begin(), extents.end());
+  // Find the number of points in the local connectivity
+  const int element_num_points =
+      alg::accumulate(extents, 1, std::multiplies<>{});
+  // Generate the connectivity data for the element
+  const std::vector<int> connectivity =
+      [&extents, &total_points_so_far ]() noexcept {
+    std::vector<int> local_connectivity;
```

Comment:
Could you add a comment: `// Possible optimisation: local_connectivity.reserve(BLAH) if we can figure out size without computing all the connectivities`